packages/pn-personagiuridica-webapp/src/components/IntegrazioneApi/PublicKeys.tsx

```diff
@@ -176,7 +185,7 @@ const PublicKeys: React.FC = () => {
             closeModalHandler={handleCloseModal}
             actionButtonLabel={t('rotate-button')}
             buttonIcon={<Sync fontSize="small" sx={{ mr: 1 }} />}
-            // actionHandler={() => apiKeyRotated(modal.apiKey?.id)}
+            actionHandler={() => handleGeneratePublicKey(modal.publicKey?.kid as string)}
```

Comment:
instead of using as string, do a check into handleGeneratePublicKey. if modal.publicKey?.kid is undefined return without do anything